cryptopan/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+//! `CryptoPAn`
+//! Anonymizes IP addresses using the CryptoPAN algorithm tightly based on the
+//! GO implementation by Yawning Angel [Repo](https://github.com/Yawning/cryptopan), which is based on the
+//! original reference implementation [paper by J. Fan, J. Xu, M. Ammar, and S. Moon.
+//! [Paper](https://ieeexplore.ieee.org/abstract/document/1181415)]  
+//!   
+//! Instantiate using a 256-bit key using `new` or `new_from_slice` and use `encrypt_v4` and `encrypt_v6` to encrypt IP addresses
+//! ```
+//! use cryptopan::CryptoPAn;
+//! use std::net::{Ipv4Addr,Ipv6Addr};
+//! use std::str::FromStr;
+//! use aes::cipher::KeyInit;
+//!   
+//! const KEY: [u8; 32] = [21, 34, 23, 141, 51, 164, 207, 128, 19, 10, 91, 22, 73, 144, 125, 16,  
+//! 216, 152, 143, 131, 121,121, 101, 39, 98, 87, 76, 45, 42, 132, 34, 2,];
+//!   
+//! let cp = CryptoPAn::new(&KEY.into());
+//! cp.encrypt_v4(Ipv4Addr::from_str("127.0.0.1").unwrap());
+//! cp.encrypt_v6(Ipv6Addr::from_str("2001:db8::1").unwrap());
+//! ```
+use aes::cipher::BlockEncryptMut;
+use aes::cipher::{generic_array::GenericArray, InvalidLength, KeyInit};
+use bitvec::field::BitField;
+use bitvec::prelude::Msb0;
+use bitvec::{bitvec, vec::BitVec};
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+/// The CryptoPAN anonymizer
+pub struct CryptoPAn {
```

Comment:
Per [RFC 430](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case) I think this should just be `CryptoPan`:

```suggestion
pub struct CryptoPan {
```